hydromt/workflows/region.py

```diff
@@ -0,0 +1,287 @@
+    kwargs = region.copy()
+    # NOTE: the order is important to prioritize the arguments
+    options = {
+        "basin": ["basid", "geom", "bbox", "xy"],
+        "subbasin": ["geom", "bbox", "xy"],
+        "interbasin": ["geom", "bbox", "xy"],
+        "geom": ["geom"],
+        "bbox": ["bbox"],
+        "mesh": ["UgridDataArray"],
+        "grid": ["raster"],
+    }
+    kind = next(iter(kwargs))  # first key of region
+    value0 = kwargs.pop(kind)
+
+    if kind in PLUGINS.model_plugins:
+        model_class = PLUGINS.model_plugins[kind]
+        other_model = model_class(root=value0, mode="r", logger=logger)
+        value0 = other_model.region
+        kwargs = dict(geom=other_model.region)
+        kind = "geom"
+    elif kind == "mesh":
+        if isinstance(value0, (str, Path)) and isfile(value0):
+            kwarg = dict(mesh=xu.open_dataset(value0))
+        elif isinstance(value0, (xu.UgridDataset, xu.UgridDataArray)):
+            kwarg = dict(mesh=value0)
+        elif isinstance(value0, (xu.Ugrid1d, xu.Ugrid2d)):
+            kwarg = dict(
+                mesh=xu.UgridDataset(value0.to_dataset(optional_attributes=True))
+            )
+        else:
+            raise ValueError(
+                f"Unrecognized type {type(value0)}."
+                "Should be a path, data catalog key or xugrid object."
+            )
+        kwargs.update(kwarg)
+    elif kind not in options:
+        k_lst = '", "'.join(list(options.keys()))
+        raise ValueError(f'Region key "{kind}" not understood, select from "{k_lst}"')
+
+    kwarg = _parse_region_value(value0, data_catalog=data_catalog)
+    if len(kwarg) == 0 or next(iter(kwarg)) not in options[kind]:
+        v_lst = '", "'.join(list(options[kind]))
+        raise ValueError(
+            f'Region value "{value0}" for kind={kind} not understood, '
+            f'provide one of "{v_lst}"'
+        )
+    kwargs.update(kwarg)
+
+    if kind in ["basin", "subbasin", "interbasin"]:
+        # retrieve global hydrography data (lazy!)
+        assert data_catalog is not None
+        assert hydrography_fn is not None
+        assert basin_index_fn is not None
+        ds_org = data_catalog.get_rasterdataset(hydrography_fn)
+        if "bounds" not in kwargs:
+            kwargs.update(basin_index=data_catalog.get_source(basin_index_fn))
+        # get basin geometry
+        geom, _ = get_basin_geometry(ds=ds_org, kind=kind, logger=logger, **kwargs)
+        _update_crs(geom, crs)
+    elif kind == "bbox":
+        bbox = kwargs["bbox"]
+        geom = gpd.GeoDataFrame(geometry=[box(*bbox)], crs=4326)
+        _update_crs(geom, crs)
+    elif kind == "geom":
+        geom = kwargs["geom"]
+        if geom.crs is None:
+            raise ValueError('Model region "geom" has no CRS')
+        _update_crs(geom, crs)
+    elif kind == "grid":
+        assert data_catalog is not None
+        ds = data_catalog.get_rasterdataset(value0, kwargs=kwargs)
+        assert ds is not None
```

Comment:
No mypy needed here the NoDataStrategy will take care of this

Reply:
mypy doesn't know about NoDataStrategy and cannot infer the raise of the error.